graphql/src/store/query.rs

```diff
@@ -360,27 +396,145 @@ fn list_values(value: Value, filter_type: &str) -> Result<Vec<Value>, QueryExecu
     }
 }
 
+enum OrderByValue {
+    Direct(String),
+    Child(String, String),
+}
+
+fn parse_order_by(enum_value: &String) -> OrderByValue {
+    let mut parts = enum_value.split("__");
+    let first = parts.next().expect("some dark magic happened");
```

Comment:
Since this is called from `build_order_by` which already returns `Result<_,_>`, make this return a `Result`, too, so we don't have a panic when something goes wrong, but rather an error that gets sent back to the user. (Though I also don't see how `next()` could be `None` here).

The general problem with `expect` and `unwrap` is that they cause panics which are much harder on the system to handle than an error, making it possible to cause trouble by sending really badly behaving queries.

Reply:
Solved in 4f1adae